From e845aca2e44eb8f9562adec07fd02d993d990ce2 Mon Sep 17 00:00:00 2001
From: Vinu Vaghasia <vinu.vaghasia@amd.com>
Date: Thu, 6 Jan 2022 16:54:46 -0600
Subject: [PATCH 1/1] ARM:dts:aspeed: Add Onyx dts adc121c and tmp468 sensors

adc121c and tmp468 sensor nodes are added with Onyx dts.

Signed-off-by: Vinu Vaghasia <vinu.vaghasia@amd.com>
---
 arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts | 41 +++++++++++++++++++----
 1 file changed, 35 insertions(+), 6 deletions(-)

diff --git a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
index 18a7d4773711..7e4057a90e20 100644
--- a/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
+++ b/arch/arm/boot/dts/aspeed-bmc-amd-onyx.dts
@@ -55,6 +55,18 @@
 			reg = <0xE0000000 0x00001000>; /* 4K */
 		};
 	};
+
+	iio-hwmon-adc121c {
+		compatible = "iio-hwmon";
+		oemname0 = "FPGA_VDD_CORE";
+		oemname1 = "FPGA_VDD_18";
+		oemname2 = "FPGA_VDD_33";
+		oemname3 = "LOM_VDD_33";
+		oemname4 = "LOM_VDD_12";
+		oemname5 = "VDD_12_DUAL";
+		io-channels = <&adc0 0>, <&adc1 0>, <&adc2 0>, <&adc3 0>,
+					<&adc4 0>, <&adc5 0>;
+	};
 };

 &mdio0 {
@@ -476,29 +488,46 @@
 			#size-cells = <0>;
 			reg = <6>;

-			adc121c@50 {
+			adc0: fpga_vdd_core@50 {
 				compatible = "ti,adc121c";
 				reg = <0x50>;
+				label = "fpga_vdd_core";
 			};
-			adc121c@51 {
+			adc1: fpga_vdd_18@51 {
 				compatible = "ti,adc121c";
 				reg = <0x51>;
+				label = "fpga_vdd_18";
 			};
-			adc121c@52 {
+			adc2: fpga_vdd_33@52 {
 				compatible = "ti,adc121c";
 				reg = <0x52>;
+				label = "fpga_vdd_33";
 			};
-			adc121c@53 {
+			adc3: lom_vdd_33@53 {
 				compatible = "ti,adc121c";
 				reg = <0x53>;
+				label = "lom_vdd_33";
 			};
-			adc121c@54 {
+			adc4: lom_vdd_12@54 {
 				compatible = "ti,adc121c";
 				reg = <0x54>;
+				label = "lom_vdd_12";
 			};
-			adc121c@55 {
+			adc5: vdd_12_dual@55 {
 				compatible = "ti,adc121c";
 				reg = <0x55>;
+				label = "vdd_12_dual";
+			};
+		};
+
+		i2c@7 {
+			#address-cells = <1>;
+			#size-cells = <0>;
+			reg = <7>;
+
+			tmp468@48 {
+				compatible = "ti,tmp468";
+				reg = <0x48>;
 			};
 		};
 	};
--
2.17.1

